### PR TITLE
Plugins: better rollbacks when 'abot plugin install' fails

### LINKED ISSUE #98: Better rollbacks when `abot plugin install` fails
Installing plugins can fail for any number of reasons (lack of internet connection, other site isn't up, a typo in your dependencies, etc.), but right now we just error out immediately. If you have an incomplete `plugins.go` file as a result, compiling will fail.

We should handle errors in `installPlugins()` like so:

```go
if err := installPlugins(); err != nil { ... }
```

 and delete the `plugins.go` file we created.

### PR DESCRIPTION
**Pull request by awwalker**
This fixes #98. Please let me know if this is headed in the right direction/what needs to be changed!

### CODE REVIEW

File: abot.go [hunk 1]
```diff
@@ -304,15 +312,15 @@ func installPlugins() {
...
+		return err
 	}
 
 	// Sync each of them to get dependencies
 	var wg sync.WaitGroup
 	wg.Add(len(plugins.Dependencies))
 	rand.Seed(time.Now().UTC().UnixNano())
 	for url, version := range plugins.Dependencies {
-		go func(url, version string) {
+		go func(url, version string) error {
```
Comment (egtann) [inline abot.go]: My understanding is that each goroutine will overwrite the returned `error`, so in this case we'll be left with whichever value the final goroutine set error equal to.

To get around this, we can use channels like this example (http://www.atatus.com/blog/goroutines-error-handling/). We can send the errors as messages, which can then each be handled as they come into the channel.

File: abot.go [hunk 2]
```diff
@@ -71,7 +71,15 @@ func main() {
 					Aliases: []string{"i"},
 					Usage:   "download and install plugins listed in plugins.json",
 					Action: func(c *cli.Context) {
-						installPlugins()
+						if err := installPlugins(); err != nil {
```
Comment (egtann) [inline abot.go]: We'd want to delete the `plugins.go` file that we created in `installPlugins` in this err block. We should also check the value returned from `os.Remove("plugins.go")` to ensure we ignore `err == os.ErrNotExist`, as that's fine (it simply means we errored out before the file was created).

**Comment by egtann** [comment]
Hey @awwalker, thanks for your help! The PR looks mostly good, but I left a few comments on it where I think we can improve.

One of the things that I'm trying to do with this project is maintain a really clean and consistent `git log`. Along those lines, could we change your commit message to follow the same formatting as the examples here? https://github.com/itsabot/abot/commits/master

Thanks!

**Comment by awwalker** [comment]
Thanks for the review @egtann I'll try and get this PR fixed over the next couple of days.

**Comment by egtann** [comment]
Hey @awwalker. Is this all set for review?

**Comment by awwalker** [comment]
Yes @egtann if you could give this another look over that'd be great. I think I have the channel working the way you envisioned for the goroutines. Let me know what you think

**Comment by egtann** [comment]
This is a great start. Thank you so much for the contribution! Merging...

**Pull request merged**